pkg/ds/farm.go

```diff
@@ -286,13 +289,16 @@ func (dsf *Farm) handleDSChanges(changes dsstore.WatchedDaemonSets, quitCh <-cha
 
 			if _, ok := err.(consulutil.AlreadyLockedError); ok {
 				if _, ok := dsf.children[dsFields.ID]; !ok {
```

Comment:
same here re: don't need the check if things are working properly. In fact, if you _did_ have that daemon set as a child it would be prudent to remove it if another farm holds it